src/Bridge/Doctrine/EventListener/PublishMercureUpdatesListener.php

```
@@ -88,6 +89,17 @@ public function __construct(ResourceClassResolverInterface $resourceClassResolve
         $this->graphQlSubscriptionManager = $graphQlSubscriptionManager;
         $this->graphQlMercureSubscriptionIriGenerator = $graphQlMercureSubscriptionIriGenerator;
         $this->reset();
+
+        $rawurlencode = ExpressionFunction::fromPhp('rawurlencode', 'escape');
+        $this->expressionLanguage->addFunction($rawurlencode);
```

As I see expression language is optional so this code will fail without it

---

That's true, I'll fix it. Thanks.

---

Yes it should throw an exception if the component is not installed indeed. Would you do it?

---

@alanpolain We have two alternatives here, I can do an if and not add the functions if there is no Expression, or make it mandatory, raising the exception. Which would we choose?

---

I think the best approach would be to not add the functions, and then throw an exception if the user uses them. WDYT?

---

It can be, I'll create the PR

---

https://github.com/api-platform/core/pull/4341